alica_tests/src/test/test_alica_scheduling.cpp

```diff
@@ -1,256 +1,130 @@
-    ASSERT_TRUE(behAAA->isFailure());
-    ASSERT_FALSE(wm->behAAASetFailureFailed);
-
-    wm->planA2PlanB = true;
-    wm->planB2PlanA = false;
-    ac->stepEngine();
-    wm->behAAASetSuccess = false;
-    wm->behAAASetFailure = false;
-
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    ASSERT_TRUE(wm->behAAASuccessInTerminate);
-    ASSERT_TRUE(wm->behAAAFailureInTerminate);
-
-    wm->planA2PlanB = false;
-    wm->planB2PlanA = true;
-    ac->stepEngine();
-
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    behAAA = alica::test::Util::getBasicBehaviour(ae, 1629895901559);
-    ASSERT_FALSE(wm->behAAASuccessInInit);
-    ASSERT_FALSE(wm->behAAAFailureInInit);
-    ASSERT_FALSE(behAAA->isSuccess());
-    ASSERT_FALSE(behAAA->isFailure());
-
-    wm->behAAASetSuccess = true;
-    SLEEP_UNTIL(behAAA->isSuccess());
-    ASSERT_TRUE(behAAA->isSuccess());
-    ASSERT_FALSE(behAAA->isFailure());
-    ASSERT_FALSE(wm->behAAASetSuccessFailed);
-    wm->behAAABlockRun = true;
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    wm->planA2PlanB = true;
-    wm->planB2PlanA = false;
-    wm->behAAABlockRun = false;
-    ac->stepEngine();
-    ASSERT_TRUE(wm->behAAASuccessInTerminate);
-    ASSERT_TRUE(wm->behAAAFailureInTerminate);
+    // Step until the test plan succeeds
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->isSuccess(plan));
 }
 
-TEST_F(AlicaSchedulingPlan, behaviourRunCheck)
+TEST_F(SingleAgentTestFixture, execOrderTransitionBetweenPlansTest)
 {
-    CounterClass::called = -1;
-    ae->start();
-
-    std::shared_ptr<alica_test::SchedWM> wm = LockedBlackboardRW(ac->editGlobalBlackboard()).get<std::shared_ptr<alica_test::SchedWM>>("worldmodel");
-    wm->execOrderTest = true;
-    ac->stepEngine();
-
-    for (int i = 0; i < 10; ++i) {
-
-        STEP_UNTIL(alica::BehAAA::runCount >= 1);
-        SLEEP_UNTIL(alica::BehAAA::runCount >= 10);
-        ASSERT_GE(alica::BehAAA::runCount, 10);
-
-        wm->planA2PlanB = true;
-        wm->planB2PlanA = false;
-        STEP_UNTIL(alica::BehAAA::runCount == 0);
-
-        ASSERT_EQ(alica::BehAAA::runCount, 0);
-
-        STEP_UNTIL(alica::BehBAA::runCount >= 1);
-        SLEEP_UNTIL(alica::BehBAA::runCount >= 10);
-        ASSERT_GE(alica::BehBAA::runCount, 10);
-
-        wm->planA2PlanB = false;
-        wm->planB2PlanA = true;
-        STEP_UNTIL(alica::BehBAA::runCount == 0);
-        ASSERT_EQ(alica::BehBAA::runCount, 0);
+    // Checks if scheduling of init, run and terminate when transitioning between 2 plans is working
+
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "SchedulingTestState")) << _tc->getLastFailure();
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->getActivePlan("SchedulingTestPlan"));
+    ASSERT_TRUE(_tc->setTransitionCond("SchedulingTestPlan", "ChooseTestState", "PlanAState")) << _tc->getLastFailure();
```

Comment:
would be better to have a sub plan for this test case. The idea is to have 1 plan for each test so its clear to anyone reading the code